src/Behat/Mink/WebAssert.php

```diff
@@ -590,11 +594,14 @@ public function fieldValueNotEquals($field, $value, TraversableElement $containe
     {
         $node   = $this->fieldExists($field, $container);
         $actual = $node->getValue();
-        $regex  = '/^'.preg_quote($value, '/').'$/ui';
 
-        $message = sprintf('The field "%s" value is "%s", but it should not be.', $field, $actual);
+        $message = sprintf(
+            'The field "%s" value is "%s", but it should not be.',
+            $field,
+            (is_array($actual) ? '['.implode(', ', $actual).']' : $actual)
+        );
 
-        $this->assert(!preg_match($regex, $actual), $message);
+        $this->assert($actual !== $value, $message);
```

Comment:
`preg_quote(false, '/')` returns `''`, `preg_quote(true, '/')` returns `'1'`.

These are the only two cases I found which might not work any longer. But these values are returned by checkboxes and these are most likely to be checked by the function WebAssert::checkboxChecked and WebAssert::checkboxNotChecked

I do not see, where else backwards compatibility could be broken.
- IMHO, there is no difference between the outcome of `'foo' === 'foo'` and `preg_match('^foo$', 'foo')`.
- `123 === '123'` won't work any more as well, but integers aren't returned by fields (at least according to PHPDoc).
- `preg_quote(['foo', 'bar'], '/')` does not work and throws an error. This is fixed now.